Client/Frontend/Fakespot/Views/FakespotAdView.swift

```diff
@@ -0,0 +1,320 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import UIKit
+import Common
+import Shared
+import ComponentLibrary
+
+// MARK: View Model
+struct FakespotAdViewModel {
+    let title: String = .Shopping.AdCardTitleLabel
+    let footerText: String = .localizedStringWithFormat(.Shopping.AdCardFooterLabel,
+                                                        FakespotName.shortName.rawValue)
+    let titleA11yId: String = AccessibilityIdentifiers.Shopping.AdCard.title
+    let cardA11yId: String = AccessibilityIdentifiers.Shopping.AdCard.card
+    let priceA11yId: String = AccessibilityIdentifiers.Shopping.AdCard.price
+    let productTitleA11yId: String = AccessibilityIdentifiers.Shopping.AdCard.productTitle
+    let descriptionA11yId: String = AccessibilityIdentifiers.Shopping.AdCard.description
+    let footerA11yId: String = AccessibilityIdentifiers.Shopping.AdCard.footer
+
+    private let tabManager: TabManager
+    var dismissViewController: (() -> Void)?
+    let productAdsData: ProductAdsData
+
+    // MARK: Init
+    init(tabManager: TabManager = AppContainer.shared.resolve(),
+         productAdsData: ProductAdsData) {
+        self.tabManager = tabManager
+        self.productAdsData = productAdsData
+    }
+
+    // MARK: Tap Product
+    func onTapProductLink() {
+        tabManager.addTabsForURLs([productAdsData.url], zombie: false, shouldSelectTab: true)
+        dismissViewController?()
```

Comment:
Dismissing the view controller is not working. When I tab on the link the bottom sheet stays open.

Reply:
View model integration was removed so we don't use mock data. It will be added on the API call.